frontend/tests/playwright/product/cross_sell.spec.ts

```diff
@@ -9,15 +9,12 @@ test.describe('Cross-sell test', () => {
       page,
    }) => {
       const products = await page.getByTestId('cross-sell-item');
-      const count = await products.count();
 
       let currentProductTitle = null;
       let currentProductPrice = null;
       const otherProductTitles = [];
 
-      for (let i = 0; i < count; i++) {
-         const product = products.nth(i);
-
+      for (const product of await products.all()) {
```

Comment:
Can we use the `forEach` call on `locator.all()`?

```
products.all().forEach()
```

Reply:
I think we can use `.forEach()` like the diff below, but now we're getting additional type errors, so we might need to cast them 😓  
<img width="634" alt="Screen Shot 2023-01-09 at 4 45 12 PM" src="https://user-images.githubusercontent.com/58952979/211436568-85f0c79e-f401-46a9-be2c-f8ab4938b5e5.png">

```diff
diff --git a/frontend/tests/playwright/product/cross_sell.spec.ts b/frontend/tests/playwright/product/cross_sell.spec.ts
index c9e52165..534d32d8 100644
--- a/frontend/tests/playwright/product/cross_sell.spec.ts
+++ b/frontend/tests/playwright/product/cross_sell.spec.ts
@@ -12,9 +12,9 @@ test.describe('Cross-sell test', () => {
 
       let currentProductTitle = null;
       let currentProductPrice = null;
-      const otherProductTitles = [];
+      const otherProductTitles: any = [];
 
-      for (const product of await products.all()) {
+      (await products.all()).forEach(async product => {
          const isCurrent = await product
             .locator('span', { hasText: 'Current item' })
             .isVisible();
@@ -26,7 +26,7 @@ test.describe('Cross-sell test', () => {
             currentProductTitle = await product
                .getByTestId('cross-sell-item-title')
                .textContent();
-            continue;
+            return;
          }
 
          // Deselect other cross-sell products
@@ -34,7 +34,7 @@ test.describe('Cross-sell test', () => {
             await product.getByTestId('cross-sell-item-title').textContent()
          );
          await product.getByTestId('cross-sell-item-select').click();
-      }
+      });
 
       // Assert total price matches price of product as it's the only one selected
       await expect(
@@ -48,7 +48,7 @@ test.describe('Cross-sell test', () => {
          .getByTestId('cart-drawer-body')
          .textContent();
 
-      otherProductTitles.forEach((otherProductTitle) => {
+      (otherProductTitles as any).forEach((otherProductTitle: any) => {
          expect(cartDrawerText).not.toContain(otherProductTitle);
       });
       expect(cartDrawerText).toContain(currentProductTitle);
```

Reply:
It is best to not use `forEach` with asynchronous tasks because the `async/await` is not respected.

https://gist.github.com/joeytwiddle/37d2085425c049629b80956d3c618971